include/mbgl/style/layers/layer.hpp.ejs

```diff
@@ -43,17 +43,17 @@ public:
 
 <% for (const property of layoutProperties) { -%>
     static <%- propertyValueType(property) %> getDefault<%- camelize(property.name) %>();
-    <%- propertyValueType(property) %> get<%- camelize(property.name) %>() const;
-    void set<%- camelize(property.name) %>(<%- propertyValueType(property) %>);
+    const <%- propertyValueType(property) %>& get<%- camelize(property.name) %>() const;
+    void set<%- camelize(property.name) %>(const <%- propertyValueType(property) %>&);
```

Comment:
Would it be better to pass by value and then move to avoid copying?
`Layer::setLayoutProperty` would create prop value and then move it to designated setter?

Reply:
Good catch, for some reason we're not using move semantics in style layers setters :-/  Might worth being done in a separate PR.

Reply:
Maybe, I can try; I made this whole change to avoid triggering the variant copy code path, which causes clang-tidy issues. See https://github.com/mapbox/variant/issues/158